st2-ask/cli-approve.txt

```diff
@@ -0,0 +1,33 @@
+vagrant@st2vagrant:~$ cat simple_approval.yaml
+---
+id: 59023cfc02ebd51154291652
+data:
+  approve: yes
+
+
+vagrant@st2vagrant:~$ st2 approval create simple_approval.yaml
++-------------+--------------------------------------------------------------+
+| Property    | Value                                                        |
++-------------+--------------------------------------------------------------+
+| id          | 59023cfc02ebd51154291652                                     |
+| data        | {                                                            |
+|             |     "approve": "yes"                                         |
+|             | }                                                            |
++-------------+--------------------------------------------------------------+
+
+vagrant@st2vagrant:~$ cat 2FA_approval.yaml
+---
+id: 59023cfc02ebd51154291653
+data:
+  second_factor: 012345
+
+
+vagrant@st2vagrant:~$ st2 approval create 2FA_approval.yaml
++-------------+--------------------------------------------------------------+
+| Property    | Value                                                        |
++-------------+--------------------------------------------------------------+
+| id          | 59023cfc02ebd51154291653                                     |
+| data        | {                                                            |
+|             |     "second_factor": "012345"                                |
+|             | }                                                            |
++-------------+--------------------------------------------------------------+
```

Comment:
How about something like the following that prompts user for information over command line?  The use of `st2 respond` covers a wider range of use cases (as compared to `st2 approval` which only covers the approval use case).  The text `Please enter passcode for the second factor authentication` can be from the description in the schema.

```
vagrant@st2vagrant:~$ st2 respond <st2-ask-execution-id>
Please enter passcode for the second factor authentication: <user input data here, don't display if it's password>
Data for st2.ask is received and validated. Resuming execution <execution-id>...
vagrant@st2vagrant:~$
```